### ISSUE TITLE
False positives on load_all_libs test

### ISSUE THREAD

**Issue by dennisklein** [FairRootGroup/FairRoot#1002]
```
test 3
      Start  3: load_all_libs

3: Test command: /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/examples/simulation/Tutorial1/macros/load_all_libs.sh
3: Test timeout computed to be: 60
3:    ------------------------------------------------------------
3:   | Welcome to ROOT 6.18/04                  https://root.cern |
3:   |                               (c) 1995-2019, The ROOT Team |
3:   | Built for linuxx8664gcc on Sep 11 2019, 15:38:23           |
3:   | From tags/v6-18-04@v6-18-04                                |
3:   | Try '.help', '.demo', '.license', '.credits', '.quit'/'.q' |
3:    ------------------------------------------------------------
3: 
3: /bin/bash: /tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib/libtinfo.so.6: no version information available (required by /bin/bash)
3: 
3: Processing /tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/examples/simulation/Tutorial1/macros/load_all_libs.C()...
3: Error in <TUnixSystem::FindDynamicLibrary>: libFairRutherford[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libFairTestDetector[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libGConfig[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libMCStack[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libMbsTutorial[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libPassive[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libPixel[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libPixelAlt[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libPropagatorExample[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libSimMQ[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libTutorial1[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libTutorial2[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: Error in <TUnixSystem::FindDynamicLibrary>: libTutorial4[.so | .dll | .dylib | .sl | .dl | .a] does not exist in /tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/FairRoot/build/lib:/tmp/fairsoft_ctest.1X6EpE/spSF4V/fairsoft/lib:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/.singularity.d/libs:.:/tmp/fairsoft_ctest.1X6EpE/install-tree/linux-ubuntu20.04-zen/gcc-9.3.0/root-6.18.04-44t3b76wmytr755txx4pilihuhz6ziwn/lib:/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/x86_64/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/x86_64/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/x86_64/x86_64:/lib/tls/x86_64:/lib/tls/x86_64:/lib/tls:/lib/x86_64/x86_64:/lib/x86_64:/lib/x86_64:/lib:/usr/lib/tls/x86_64/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/x86_64/x86_64:/usr/lib/x86_64:/usr/lib/x86_64:/usr/lib
3: 
3: 
3: Macro finished succesfully.
 3/54 Test  #3: load_all_libs ..........................   Passed    1.59 sec
```

This is fixed in #933.

**Comment by ChristianTackeGSI** [comment]
The issue is fully correct: The test succeeds always (unless root is broken).

I have not yet looked deeply enough, why some of the example libraries are not loadable.
Dennis' cmake modernization branch has a fix for it. I can extract it.

But I do not fully understand, what this test is actually checking?
* Is `gSystem->Load()` actually an API, that we want people to use on our libraries?!
* I assume, we test our libraries through more higher level tests anyhow? So all of those would fail as well, if the library loading is broken in some way? So is this test trying to help pinpoint issues more clearly?

**Comment by fuhlig1** [comment]
Actually the test should fail if one of the libraries can't be loaded. For me it is unclear why this isn't the case. 

What we do in CBM is to load each library on their own. This allows to find issues with missing dependencies which may not show up when running a macro. When running a macro normally many libraries are loaded and it is very likely that one of them already loads the needed dependencies.

`gSystem->Load()`is the API which must be used in older times before the `rootmap` files were available which moves the explicit loading of libraries behind the scenes.

**Comment by ChristianTackeGSI** [comment]
@fuhlig1, many thanks for explaining the reasoning behind the test!

> Actually the test should fail if one of the libraries can't be loaded. For me it is unclear why this isn't the case.

The return value of `gSystem->Load()` isn't checked. As simple as that.

> `gSystem->Load()` is the API which must be used in older times before the rootmap files were available which moves the explicit loading of libraries behind the scenes.

That sounds like "We do not need to test `gSystem->Load()` as a public API endpoint for FairRoot".
So from that point the test is not needed (any longer).

> What we do in CBM is to load each library on their own. This allows to find issues with missing dependencies which may not show up when running a macro. When running a macro normally many libraries are loaded and it is very likely that one of them already loads the needed dependencies.

That's an interesting approach!
I think, it is a very good goal to make all libraries load their dependencies properly. AFAIK, FairRoot currently already achieves that goal.

The current test loads all libraries in order. So it is not actually a good test for this. Loading each library alone would be better, as you already suggest. Maybe we should use something like in CBM? But read on.

On Linux machines, we can let the linker test this during build time for us. See #1082.

So I propose we drop the `load_all_libs` test alltogether and use the approach from #1082. If we later on notice, that this is not enough (for whatever reason), we should look at CBM's approach.

**Comment by fuhlig1** [comment]
@ChristianTackeGSI, 

I completely agree with your proposal to let the linker test the completeness of the dependencies on build time. On macosx this is even the default setting of the linker.
We did not use the feature up to now since without our CMake modernisation you would need to know the complete dependency tree of your direct dependencies. This was something we tried to avoid.

**Comment by dennisklein** [comment]
resolved in 906f6e8493284f083b3d30a50b9001cce5109007
